controllers/skills.js

```diff
@@ -0,0 +1,70 @@
+const skills = require("../models/skills");
+const logger = require("../utils/logger");
+
+/**
+ * Award a skill
+ *
+ * @param req {Object} - Express request object
+ * @param req.body {Object} - User object
+ * @param res {Object} - Express response object
+ */
+async function awardSkill(req, res) {
+  try {
+    const { username } = req.params;
+    const addedData = await skills.awardSkill(req.body, username);
+    return res.json({
+      message: "Added data successfully!",
+      content: addedData,
+    });
+  } catch (error) {
+    logger.error("Error posting skill data: ", error);
+    return res.boom.notFound("User doesn't exist");
```

Comment:
If there is any error we'll be sending a 404 response
Is this the intended behaviour?

Reply:
yes

Reply:
Can there be any other cause of error other than `user not found`. Even in that case should be return 404 or 400 since 404 is meant when the requested resource is not found. Here we are not requesting for a resource

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404

Reply:
1. I think there can be an error response for no data(empty object) in req.body as well
2. 400 status code would be better since there is no req for resource...